dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -1942,7 +1969,8 @@ void NO_INLINE Aggregator::mergeWithoutKeyStreamsImpl(
     AggregatedDataWithoutKey & res = result.without_key;
     if (!res)
     {
-        AggregateDataPtr place = result.aggregates_pool->alloc(total_size_of_aggregate_states);
+        AggregateDataPtr place = result.aggregates_pool->alloc_align(total_size_of_aggregate_states,
```

Comment:
Also look at `SpecializedAggregator` that is used when `compile` setting is enabled.